### PR TITLE
safety: sample_t is not fully reset on safety mode init

### PR DESCRIPTION
**Pull request by sshane**
The safety mode set just sets the min and max, it doesn't clear out the values array, so on rx of one new measurement, it will use the previously stored values to update the sample_t min and max

this is just a test to catch it, moving the base test to https://github.com/commaai/panda/pull/1250

### CODE REVIEW

File: board/safety.h [hunk 1]
```diff
@@ -345,6 +345,12 @@ int set_safety_hooks(uint16_t mode, uint16_t param) {
   valid_steer_req_count = 0;
   invalid_steer_req_count = 0;
 
+  for (int i = 0; i < MAX_SAMPLE_VALS; i++) {
+    torque_meas.values[i] = 0;
+    torque_driver.values[i] = 0;
+    angle_meas.values[i] = 0;
+  }
+
```
Comment (sshane) [inline board/safety.h:353]: can also use memset?

File: tests/safety/common.py [hunk 2]
```diff
@@ -634,6 +638,8 @@ def test_reset_angle_measurements(self):
     # reset sample_t by reinitializing the safety mode
     self.setUp()
 
+    # rx one to update sample_t
+    self.assertTrue(self._rx(self._angle_meas_msg(0)))
```
Comment (adeebshihadeh) [inline tests/safety/common.py:642]: why is the RX necessary?

File: tests/safety/common.py [hunk 3]
```diff
@@ -634,6 +638,8 @@ def test_reset_angle_measurements(self):
...
     self.assertEqual(self.safety.get_angle_meas_min(), 0)
     self.assertEqual(self.safety.get_angle_meas_max(), 0)
```
Comment (adeebshihadeh) [inline tests/safety/common.py:644]: add an assert before the safety mode reset that it's not the reset value

File: board/safety.h (hunk 1 above)
Comment (adeebshihadeh) [reply board/safety.h:353]: for loop is more readable

File: tests/safety/common.py (hunk 2 above)
Comment (sshane) [reply tests/safety/common.py:763]: because receiving a message calls `update_sample` which sets the min and max from the non-zero sample array

Comment (adeebshihadeh) [reply tests/safety/common.py:763]: that should also be reset on safety mode init

Comment (sshane) [reply tests/safety/common.py:763]: That's what this does https://github.com/commaai/panda/pull/1376/files#diff-9e71d0935391e66564e1fe068f0e9c0b7a2523e5a31c96d2556205ee60b6d15aR347-R352

Comment (sshane) [reply tests/safety/common.py:763]: done a little cleaner

**Pull request merged**